### PR TITLE
Fixed various type punning warnings by using memcpy()

### PR DESCRIPTION
**Pull request by seanm**
Modern compilers optimize memcpy() very well, this will likely produce the same code.

### CODE REVIEW

File: nifti2/nifti_tool.c [hunk 1]
```diff
@@ -3849,7 +3848,7 @@ int modify_field(void * basep, field_s * field, const char * data)
 
          case DT_INT8:
          {
-            max = 127;
+            max = UINT8_MAX;
```
Comment (issakomi) [inline nifti2/nifti_tool.c:3851]: Is this change intentional? 127 vs. 255. Also 32767 vs. 65535 below.

Comment (seanm) [reply nifti2/nifti_tool.c:3851]: Whoops! Excellent catch.

Comment (seanm) [reply nifti2/nifti_tool.c]: Fixed.

**Review by hjmjohnson — APPROVED** [review]

**Comment by seanm** [comment]
Thanks @hjmjohnson !

**Comment by afni-rickr** [comment]
Note that this commit has been undone with PR 170.  The addressing in 166 does not account for types in indexing.  For example `char * offsetp = &(((char *)basep + field->offset)[fc]);` with float data needs [fc] to offset by floats, not chars.

**Comment by seanm** [comment]
> Note that this commit has been undone with PR 170. The addressing in 166 does not account for types in indexing.

Oh damn, indeed you are right.  The many brackets threw me for a loop.  Disturbing that no test failed due to this.  I'll make a new PR to fix it.

**Comment by afni-rickr** [comment]
I could add a test for that, something like:
```
nifti_tool -mod_hdr -new_dim 4 64 64 21 180 0 0 0                \
            -mod_field pixdim "0.0 4.0 4.0 6.0 3.0 1.0 1.0 1.0"  \
            -prefix new_image.nii -infiles MAKE_IM
nifti_tool -disp_hdr -field pixdim -quiet -infiles new_image.nii
```
If pixdim were set using a variable, that could be checked against the output.

**Comment by seanm** [comment]
@afni-rickr please see my fix in https://github.com/NIFTI-Imaging/nifti_clib/pull/172
